### PR TITLE
Refactoring cra probe

### PR DESCRIPTION
**Pull request by lcarrazana**
- Refactoring cra probe to use simplified approach with connection obj
- Consolidating feature file.

Relates to #329

### CODE REVIEW

File: service_packs/kubernetes/container_registry_access/container_registry_access.go [hunk 1]
```diff
@@ -1,177 +1,240 @@
...
 
-func (s *scenarioState) theDeploymentAttemptIsAllowed() error {
+func (scenario *scenarioState) podCreationXWithContainerImageFromYRegistry(expectedResult, registryAccess string) error {
+	// Supported values for 'expectedResult':
+	//	'succeeds'
+	//	'is denied'
+
+	// Supported values for 'registryAccess':
+	//	'authorized'
+	//  'unauthorized'
```
Comment (eddie-knight) [inline service_packs/kubernetes/container_registry_access/container_registry_access.go:68]: spacing

Comment (lcarrazana) [inline service_packs/kubernetes/container_registry_access/container_registry_access.go:68]: Space is wrong

File: service_packs/kubernetes/container_registry_access/container_registry_access.go [hunk 2]
```diff
@@ -1,177 +1,240 @@
...
-	stepTrace.WriteString(fmt.Sprintf(
-		"Attempts to deploy a container from %s. Retains pod creation result in scenario state; ",
-		config.Vars.ServicePacks.Kubernetes.UnauthorisedContainerRegistry))
-	payload = struct {
-		UnauthorizedRegistry string
-		PodAudit             *kubernetes.PodAudit
-	}{
-		UnauthorizedRegistry: config.Vars.ServicePacks.Kubernetes.UnauthorisedContainerRegistry,
-		PodAudit:             podAudit,
+	// TODO: We are assuming too much here- if the image successfully pulls but fails to build, this will still fail
```
Comment (lcarrazana) [inline service_packs/kubernetes/container_registry_access/container_registry_access.go:110]: Remove TODO

Comment (eddie-knight) [inline service_packs/kubernetes/container_registry_access/container_registry_access.go:110]: comment can be removed now

File: service_packs/kubernetes/container_registry_access/container_registry_access.go [hunk 3]
```diff
@@ -1,177 +1,240 @@
...
-		cra = NewDefaultCRA()
-	}
+	ctx.AfterSuite(func() {
+	})
 }
 
 // ScenarioInitialize initialises the specific test steps.  This is essentially the creation of the test
 // which reflects the tests described in the events directory.  There must be a test step registered for
 // each line in the feature files. Note: Godog will output stub steps and implementations if it doesn't find
 // a step / function defined.  See: https://github.com/cucumber/godog#example.
```
Comment (eddie-knight) [inline service_packs/kubernetes/container_registry_access/container_registry_access.go:174]: this long description doesn't really need to be everywhere like it is presently... lets remove it, and we can put this into a README somewhere if we think it's helpful.

File: service_packs/kubernetes/container_registry_access/container_registry_access.go (hunk 1 above)
Comment (lcarrazana) [reply service_packs/kubernetes/container_registry_access/container_registry_access.go:68]: - [ ] Done. Replaced spaces with tab char

File: service_packs/kubernetes/container_registry_access/container_registry_access.go (hunk 2 above)
Comment (lcarrazana) [reply service_packs/kubernetes/container_registry_access/container_registry_access.go:110]: - [ ] Done

File: service_packs/kubernetes/container_registry_access/container_registry_access.go (hunk 3 above)
Comment (lcarrazana) [reply service_packs/kubernetes/container_registry_access/container_registry_access.go:174]: Replaced long description with this:
`// ScenarioInitialize provides initialization logic before each scenario is executed`

File: service_packs/kubernetes/container_registry_access/container_registry_access.go (hunk 1 above)
Comment (lcarrazana) [reply service_packs/kubernetes/container_registry_access/container_registry_access.go:68]: https://www.youtube.com/watch?v=JfEIkkDkrmE

**Review by eddie-knight — APPROVED** [review]

**Pull request merged**